Vectorize BadELF

SWeav02 opened the pull request:
This pull request aims to vectorize the various components of the BadELF algorithm. This can result in speedups of a couple orders of magnitude. This pull request will include the following:

- [x] Update partitioning plane selection
- [x] Use numpy or dask array methods for assigning voxels not split by planes
- [x] Use numpy or dask array methods for calculating fraction of voxels split by planes
- [x] Clean up comments
- [x] Test

SWeav02 committed:
Add matrix property to grid

SWeav02 committed:
Add matrix property to grid

SWeav02 committed:
Update partitioning scheme

SWeav02 committed:
Update partitioning scheme

SWeav02 committed:
Update Voxel Assignment Scheme

SWeav02 committed:
Update Voxel Assignment Scheme

SWeav02 commented:
@jacksund It's looking like there are more bugs with the new code than I was expecting so I probably won't be able to get it done by tomorrow. You can go ahead and release v0.16.0 with the old code and close issue #546. I'll let you know when this is looking better and is ready for a new release.

SWeav02 committed:
Fix bug in partitioning

SWeav02 committed:
Fix bug in partitioning

jacksund commented:
Sounds good. We can do another release whenever you're ready. 

I'll be making the v0.16.0 release tomorrow some time (I got caught up in some other things today)

jacksund commented:
also FYI -- the CI failing isn't your fault. Black received an update last night that I will fix tomorrow morning: #607

SWeav02 committed:
Small updates to voxel assignment

SWeav02 committed:
Small updates to voxel assignment

SWeav02 committed:
update indexing in partitioning

SWeav02 committed:
update indexing in partitioning

SWeav02 committed:
Add partitioning refinement

SWeav02 committed:
Add partitioning refinement

SWeav02 committed:
Fix bug with calculating voxel frac coords

SWeav02 committed:
Fix bug with calculating voxel frac coords

SWeav02 committed:
Small documentation improvement

SWeav02 committed:
Small documentation improvement

SWeav02 committed:
Small documentation improvement

SWeav02 committed:
Small documentation improvement

SWeav02 committed:
Small bug fix

SWeav02 committed:
Small bug fix

SWeav02 committed:
Fix major error in voxel assignment

For voxels split by planes, sites are found for each vertex of the voxel. Previously, sites were assigned even if the vertex sat on a plane. This caused some vertices to be assigned to multiple sites and caused an error.

SWeav02 committed:
Fix major error in voxel assignment

For voxels split by planes, sites are found for each vertex of the voxel. Previously, sites were assigned even if the vertex sat on a plane. This caused some vertices to be assigned to multiple sites and caused an error.

SWeav02 committed:
black and isort

SWeav02 committed:
black and isort

SWeav02 committed:
Update Voxel Assignment Doc Strings

SWeav02 committed:
Update Voxel Assignment Doc Strings

SWeav02 committed:
Update remaining Doc strings

SWeav02 committed:
Update remaining Doc strings

SWeav02 committed:
Remove plane reduction method

SWeav02 committed:
Remove plane reduction method

SWeav02 committed:
Fix plane selection bug in voxel assignment

SWeav02 committed:
Fix plane selection bug in voxel assignment

SWeav02 committed:
Fix bug in electride voxel assignment

SWeav02 committed:
Fix bug in electride voxel assignment

SWeav02 committed:
Merge pull request #2 from SWeav02/simple_plane_split

Remove plane splitting assignment

SWeav02 committed:
Add backup for when voxels are assigned to multiple sites

SWeav02 committed:
Add backup for when voxels are assigned to multiple sites

SWeav02 committed:
Fix bug in electride finder

SWeav02 committed:
Fix bug in electride finder

SWeav02 commented:
@jacksund As a heads up this is almost ready. I'll be testing some things a bit more tomorrow. Assuming all goes well I'll mark it ready for review.

SWeav02 committed:
Remove Core parameter

With the new numpy based badelf, I haven't implemented a core selection. This only effects the Dask arrays that are used for larger voxel grids. I can probably add the parameter back in if it appears to be important.

SWeav02 committed:
Remove Core parameter

With the new numpy based badelf, I haven't implemented a core selection. This only effects the Dask arrays that are used for larger voxel grids. I can probably add the parameter back in if it appears to be important.

SWeav02 committed:
Merge remote-tracking branch 'origin/vectorize_badelf' into vectorize_badelf

jacksund commented:
I'm not not sure what you just did with the commits haha. Normally you "rebase" (there is/was a button for it on this PR), but you somehow managed to duplicate all of the commits on main instead. If the branch is all out of wack, I may just move your badelf changes into a new MR.

Let me if/when you'd need help with this

SWeav02 commented:
Lol I tried rebasing in gitkraken and probably messed something up somewhere in the process. I'm making one more commit to suppress warnings within context and then I'm done on my end. 

I've also run into a few issues after the rebase with running workflows. Not sure if those are caused by the improper rebase or something else, but I'll send info about them once I'm done with working out the warnings.

SWeav02 committed:
supress warnings in context

SWeav02 committed:
Add BadelfConfig line when registering badelf

SWeav02 commented:
@jacksund Everything seems to be running fine outside of the error I'm running into in issue #620. Once that's fixed I'll mess around with a few different structures to make sure I didn't miss any unexpected errors and then mark this ready for review.

SWeav02 committed:
Remove cores parameter documentation

SWeav02 committed:
supress warning in context

SWeav02 commented:
Everything is working smoothly on my end. I think the tests are still failing because of black. @jacksund I can look into it tomorrow if you don't have a quick fix.

SWeav02 committed:
Merge branch 'main' into vectorize_badelf

jacksund committed:
run black

jacksund commented:
@SWeav02 you have a branch called `update_electride_search` too -- did you want that part of the MR?

jacksund commented on src/simmate/command_line/config.py:
```diff
@@ -56,6 +56,7 @@ def add(app_name: str):
     if app_name == "badelf":
         settings.add_apps_and_update(
             [
+                "simmate.apps.configs.BadelfConfig",
```
thanks for catching my typo 😄

SWeav02 commented:
@jacksund No I'm currently messing around with stuff on that branch and it won't be ready for a bit. It's mostly testing some things for a new project.

jacksund committed:
Merge branch 'main' into vectorize_badelf

jacksund commented:
sounds good 👍 I'll merge this once the CI finishes
